From d24023e375704860c6c8b91c3af3034669aa1bc5 Mon Sep 17 00:00:00 2001
Message-Id: <d24023e375704860c6c8b91c3af3034669aa1bc5.1656310671.git.geliang.tang@suse.com>
From: Rikard Falkeborn <rikard.falkeborn@gmail.com>
Date: Sat, 5 Jun 2021 22:38:05 +0200
Subject: [PATCH] platform/x86: intel_pmt_crashlog: Constify static
 attribute_group struct
Git-commit: d24023e375704860c6c8b91c3af3034669aa1bc5
Patch-mainline: v5.14-rc1
References: bsn#171
Modified-by-SEL: No

The only use of pmt_crashlog_group is to assign its address to the
attr_grp field in the intel_pmt_namespace struct, which is a pointer to
const attribute_group. Make it const to allow the compiler to put it in
read-only memory.

Signed-off-by: Rikard Falkeborn <rikard.falkeborn@gmail.com>
Link: https://lore.kernel.org/r/20210605203807.60547-3-rikard.falkeborn@gmail.com
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/platform/x86/intel_pmt_crashlog.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/x86/intel_pmt_crashlog.c b/drivers/platform/x86/intel_pmt_crashlog.c
index 92d315a16cfd..56963ceb6345 100644
--- a/drivers/platform/x86/intel_pmt_crashlog.c
+++ b/drivers/platform/x86/intel_pmt_crashlog.c
@@ -218,7 +218,7 @@ static struct attribute *pmt_crashlog_attrs[] = {
 	NULL
 };
 
-static struct attribute_group pmt_crashlog_group = {
+static const struct attribute_group pmt_crashlog_group = {
 	.attrs	= pmt_crashlog_attrs,
 };
 
-- 
2.35.3

